pkg/controller/loadtest.go

```diff
@@ -303,18 +303,42 @@ func (c *Controller) syncHandler(key string) error {
 	}
 	// copy object before mutate it
 	loadTest := loadTestFromCache.DeepCopy()
-	defer c.updateLoadTestStatus(ctx, key, loadTest, loadTestFromCache)
 
+	// check and delete stale finished/errored loadtests
+	if checkLoadTestLifeTimeExceeded(loadTest, c.cfg.CleanUpThreshold) {
+		c.logger.Info("Deleting loadtest",
+			zap.String("loadtest", loadTest.Name),
+			zap.String("phase", string(loadTest.Status.Phase)),
+		)
+		err = c.kangalClientSet.KangalV1().LoadTests().Delete(ctx, loadTest.Name, metaV1.DeleteOptions{})
+		if err != nil {
+			// The LoadTest resource may be conflicted, in which case we stop processing.
+			if errors.IsConflict(err) {
+				utilRuntime.HandleError(fmt.Errorf("there is a conflict with loadtest '%s' between datastore and cache. it might be because object has been removed or modified in the datastore", key))
```

Comment:
```suggestion
				utilRuntime.HandleError(fmt.Errorf("there is a conflict with loadtest %q between datastore and cache. it might be because object has been removed or modified in the datastore", key))
```